src/cpp/core/system/file_monitor/Win32FileMonitor.cpp

```diff
@@ -41,13 +42,15 @@ namespace {
 // buffer size for notifications (cannot be > 64kb for network drives)
 const std::size_t kBuffSize = 32768;
 
+// handle registry
+std::set<Handle> s_handleRegistry;
```

Comment:
Do accesses to this registry require mutexes? (noticed they're all bare; haven't looked too closely but I know file monitors run on a thread)

Reply:
I think you're right: these actions can be triggered either by the main thread, or by the file monitor thread itself. For example, `registerMonitor()` is typically called by the main thread, whereas `unregisterMonitor()` here is being called by the thread itself.

A race here should be unlikely since registration normally only ever happens at startup, and de-registration on quit / shutdown, but one could imagine the potential in session that quickly close / exit for some reason.